setup.py

```diff
@@ -291,56 +250,60 @@ def finalize_options(self):
-            "edgedb.datatypes.datatypes",
-            ["edgedb/datatypes/args.c",
-             "edgedb/datatypes/record_desc.c",
-             "edgedb/datatypes/tuple.c",
-             "edgedb/datatypes/namedtuple.c",
-             "edgedb/datatypes/object.c",
-             "edgedb/datatypes/set.c",
-             "edgedb/datatypes/hash.c",
-             "edgedb/datatypes/array.c",
-             "edgedb/datatypes/link.c",
-             "edgedb/datatypes/linkset.c",
-             "edgedb/datatypes/repr.c",
-             "edgedb/datatypes/comp.c",
-             "edgedb/datatypes/datatypes.pyx"],
+            extra_link_args=LDFLAGS,
+        ),
+        Extension(
+            'edgedb.datatypes.datatypes',
+            [
+                'edgedb/datatypes/args.c',
+                'edgedb/datatypes/record_desc.c',
+                'edgedb/datatypes/tuple.c',
+                'edgedb/datatypes/namedtuple.c',
+                'edgedb/datatypes/object.c',
+                'edgedb/datatypes/set.c',
+                'edgedb/datatypes/hash.c',
+                'edgedb/datatypes/array.c',
+                'edgedb/datatypes/link.c',
+                'edgedb/datatypes/linkset.c',
+                'edgedb/datatypes/repr.c',
+                'edgedb/datatypes/comp.c',
+                'edgedb/datatypes/datatypes.pyx',
+            ],
             extra_compile_args=CFLAGS,
-            extra_link_args=LDFLAGS),
-
-        distutils_extension.Extension(
-            "edgedb.protocol.protocol",
-            ["edgedb/protocol/protocol.pyx"],
+            extra_link_args=LDFLAGS,
+        ),
+        Extension(
+            'edgedb.protocol.protocol',
+            ['edgedb/protocol/protocol.pyx'],
             extra_compile_args=CFLAGS,
             extra_link_args=LDFLAGS,
-            include_dirs=INCLUDE_DIRS),
-
-        distutils_extension.Extension(
-            "edgedb.protocol.asyncio_proto",
-            ["edgedb/protocol/asyncio_proto.pyx"],
+            include_dirs=INCLUDE_DIRS,
+        ),
+        Extension(
+            'edgedb.protocol.asyncio_proto',
+            ['edgedb/protocol/asyncio_proto.pyx'],
             extra_compile_args=CFLAGS,
             extra_link_args=LDFLAGS,
-            include_dirs=INCLUDE_DIRS),
-
-        distutils_extension.Extension(
-            "edgedb.protocol.blocking_proto",
-            ["edgedb/protocol/blocking_proto.pyx"],
+            include_dirs=INCLUDE_DIRS,
+        ),
+        Extension(
+            'edgedb.protocol.blocking_proto',
+            ['edgedb/protocol/blocking_proto.pyx'],
             extra_compile_args=CFLAGS,
             extra_link_args=LDFLAGS,
-            include_dirs=INCLUDE_DIRS),
+            include_dirs=INCLUDE_DIRS,
+        ),
     ],
-    cmdclass={'build_ext': build_ext},
+    cmdclass={'sdist': sdist, 'build_py': build_py, 'build_ext': build_ext},
     test_suite='tests.suite',
     install_requires=[
-        'typing-extensions~=3.7.4',
```

Comment:
Only install `typing-extensions` on python < 3.9 as the package only uses `TypedDict` and `Literal` which are present in python >3.9